### PR TITLE
feat: migrate to the new version of Jupiter's SecurityPolicy

### PR DESCRIPTION
**Pull request by marcambier**
**Issue**

https://github.com/gravitee-io/issues/issues/7991

---
**Gravitee.io Automatic Deployment**

🚀 A prerelease version of this package has been published on Gravitee's private artifactory, you can:
 - use it directly by updating your project with version: `2.3.0-feat-jupiterPlans-SNAPSHOT`
 - download it from Artifactory [here](https://odbxikk7vo-artifactory.services.clever-cloud.com/gravitee-snapshots/io/gravitee/policy/gravitee-policy-jwt/2.3.0-feat-jupiterPlans-SNAPSHOT/gravitee-policy-jwt-2.3.0-feat-jupiterPlans-SNAPSHOT.zip)

### CODE REVIEW

**Review by jhaeyaert — APPROVED** [review]
Few comments regarding some variable names but looks good to me.
The work done on the integration tests is really impressive and definitely super mega useful bravo 👏

File: src/test/java/io/gravitee/policy/jwt/JwtPolicyIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+
+        assert401unauthorized(obs);
+    }
+
+    @Test
+    @DisplayName("Should receive 401 - Unauthorized when calling with an expired token")
+    void shouldGet401_ifExpiredToken(WebClient client) throws Exception {
+        wiremock.stubFor(get("/team").willReturn(ok("response from backend")));
+
+        String accessToken = getJsonWebToken(-50);
```
[the remark this answers is not in the document]
Comment (marcambier) [reply src/test/java/io/gravitee/policy/jwt/JwtPolicyIntegrationTest.java:137]: :+1:

File: src/main/java/io/gravitee/policy/jwt/JWTPolicy.java [hunk 2]
```diff
@@ -120,7 +118,7 @@ public Completable onMessageRequest(final MessageExecutionContext ctx) {
     }
 
     private Completable handleSecurity(final HttpExecutionContext ctx) {
-        return extractToken(ctx)
+        return extractAccessToken(ctx)
```
[the remark this answers is not in the document]
Comment (marcambier) [reply src/main/java/io/gravitee/policy/jwt/JWTPolicy.java:121]: The reason was to distinguish with  `extractSecurityToken`. But I agree, it's confusing as a JWT token is not necessarily an access token.
I turned it back to `extractToken`.

**Review by gcusnieux — APPROVED** [review]

**Pull request merged**